apps/google/sheets/models.py

```
@@ -0,0 +1,131 @@
+    def discover_from_url(
+        cls,
+        *,
+        sheet_url: str,
+        drive_account: DriveAccount | None = None,
+        is_public: bool | None = None,
+    ) -> "GoogleSheet":
+        """Create or update a tracked sheet from a URL and optional account."""
+
+        spreadsheet_id = cls.spreadsheet_id_from_url(sheet_url)
+        if not spreadsheet_id:
+            raise ValidationError({"sheet_url": _("Unable to parse spreadsheet id from URL.")})
```

Ruff (0.15.0)

[error] Local variable `_` referenced before assignment

(F823)